tests/unit_tests/test_solver.cpp

```diff
@@ -0,0 +1,59 @@
+#include "gtest/gtest.h"
+
+#include <Kokkos_Core.hpp>
+
+#include "src/rigid_pendulum_poc/solver.h"
+
+namespace oturb_tests {
+
+using namespace openturbine::rigid_pendulum;
+
+TEST(SolverType, DefaultValue) {
+    // Test that the default value of a SolverType object is kNone
+    SolverType type {};
+    EXPECT_EQ(type, SolverType::kNone);
+}
+
+TEST(SolverType, Comparison) {
+    // Test that different solver types can be compared
+    EXPECT_LT(SolverType::kNone, SolverType::kDirectLinearSolver);
+    EXPECT_EQ(SolverType::kDirectLinearSolver, SolverType::kDirectLinearSolver);
+    EXPECT_NE(SolverType::kDirectLinearSolver, SolverType::kNone);
+    EXPECT_LT(SolverType::kNone, SolverType::kIterativeLinearSolver);
+    EXPECT_EQ(SolverType::kIterativeLinearSolver, SolverType::kIterativeLinearSolver);
+    EXPECT_NE(SolverType::kIterativeLinearSolver, SolverType::kDirectLinearSolver);
+    EXPECT_NE(SolverType::kIterativeLinearSolver, SolverType::kNone);
+    EXPECT_LT(SolverType::kNone, SolverType::kIterativeNonlinearSolver);
+    EXPECT_EQ(SolverType::kIterativeNonlinearSolver, SolverType::kIterativeNonlinearSolver);
+    EXPECT_NE(SolverType::kIterativeNonlinearSolver, SolverType::kIterativeLinearSolver);
+    EXPECT_NE(SolverType::kIterativeNonlinearSolver, SolverType::kDirectLinearSolver);
+    EXPECT_NE(SolverType::kIterativeNonlinearSolver, SolverType::kNone);
+}
```

Comment:
This test puts a contraint on our enum ordering.  Isn't the usual use case just to check for equality (which is a built-in functionality, similar to checking integer equality)?  If we don't need to know the ordering, or underlying number representing the enum, I don't know if we should add this constraint yet.